modules/layout/react/lib/Column.tsx

```diff
@@ -35,19 +34,20 @@ const ColumnContainer = styled('div')<ColumnProps>(
       return;
     }
 
-    if (!spacing) {
-      return {padding: `0 ${canvas.spacing.xs}`};
-    }
-
     return {padding: `0 ${spacing}px`};
   },
-  ({columns, width}) => {
+  ({columns, width, spacing}) => {
     if (width) {
       return {width};
     }
 
     if (columns) {
-      return {width: `${(columns / 12) * 100}%`};
+      if (columns === 12) {
+        return {width: `100%`};
+      } else {
+        const compensation = ((12 - columns) / 12) * (spacing! * 2);
```

Comment:
I may be reading this wrong but doesn't this mean that `compensation` can be `NaN` if we don't define spacing?

Reply:
Javascript is weird...

```ts
1 * null // 0
```

Reply:
But I would suggest the following to not rely on this quirk:
```suggestion
        const compensation = ((12 - columns) / 12) * (spacing || 0 * 2);
```

Reply:
```ts
1 * undefined // NaN
```

Reply:
Updated! Ended up having to add in extra parentheses– 

```tsx
const compensation = ((12 - columns) / 12) * ((spacing || 0) * 2);
```

Reply:
Cool. I had parenthesis in originally and did some light testing and removed it.